### DISCUSSION TITLE
How to use the tinydream backend?

### DISCUSSION THREAD

**Discussion by ER-EPR** [mudler/LocalAI#1536] [Q&A]
I test tinydream in v2.3.1 docker image, no success error. @M0Rf30 Could you give me some help?
```
curl http://ai.aaa.bbb/v1/images/generations -H "Content-Type: application/json" -d '{"prompt": "tree", "model": "tinydream", "step": 51, "size": "512x512"}'
```
error message:
```
{"error":{"code":500,"message":"grpc process not found: /tmp/localai/backend_data/backend-assets/grpc/tinydream. some backends(stablediffusion, tts) require LocalAI compiled with GO_TAGS","type":""}}
```

**Comment by M0Rf30** [comment]
Hi,

download the model:
```sh
curl http://localhost:8080/models/apply -H "Content-Type: application/json" -d '{
    "id": "model-gallery@tinydream"
}'
```

to start the process
```sh
curl localhost:8080/v1/images/generations -H "Content-Type: application/json" -d '{
              "prompt": "floating hair, portrait, ((loli)), ((one girl)), cute face, hidden hands, asymmetrical bangs, beautiful detailed eyes, eye shadow, hair ornament, ribbons, bowties, buttons, pleated skirt, (((masterpiece))), ((best quality)), colorful|((part of the head)), ((((mutated hands and fingers)))), deformed, blurry, bad anatomy, disfigured, poorly drawn face, mutation, mutated, extra limb, ugly, poorly drawn hands, missing limb, blurry, floating limbs, disconnected limbs, malformed hands, blur, out of focus, long neck, long body, Octane renderer, lowres, bad anatomy, bad hands, text",
              "mode": 2,  "seed":9000,
              "size": "256x256", 
              "n":2,
              "model": "tinydream"
}'
```
Using a size like 520x520 (height and width > 512) will also apply the upscaler

**Reply by ER-EPR** [reply]
Is it available in quay.io/go-skynet/local-ai:v2.3.1 ? I still get this error, do I need to download something else except the model? It seems something missing in my container.
```
{"error":{"code":500,"message":"grpc process not found: /tmp/localai/backend_data/backend-assets/grpc/tinydream. some backends(stablediffusion, tts) require LocalAI compiled with GO_TAGS","type":""}}
```

I can successfully run the stablediffusion.yaml and LLMs in the model-gallery.
I have the tinydream model in the /models folder, and listed in the api.
```
curl http://abc.def/v1/models
{"object":"list","data":[{"id":"gpt4all-j","object":"model"},{"id":"stablediffusion","object":"model"},{"id":"whisper-1","object":"model"},{"id":"tinydream","object":"model"},{"id":"baichuan-7b","object":"model"},{"id":"lost+found","object":"model"}]}
```

**Reply by ER-EPR** [reply]
Thanks very much for the help, but I still can't get it to work. Which docker image support the tinydream backend? I test it with quay.io/go-skynet/local-ai:v2.3.1, but get the error above.  @M0Rf30

**Reply by M0Rf30** [reply]
Maybe that's because the GO_TAGS="tinydream" is not enabled in docker images builds. Need to check

**Reply by ER-EPR** [reply]
@M0Rf30 yes, the official dockerfile has two GO_TAGS, the second one doesn't include tinydream.
![image](https://github.com/mudler/LocalAI/assets/38782737/1245e6c0-1456-4608-8f86-a13c9c3ffa6e)
Could you please add it? I'm not familiar with github.

**Comment by ER-EPR** [comment]
I can't get it compiled into the image. Could you experts take some time to look into it? @M0Rf30 @mudler

**Comment by M0Rf30** [comment]
> I can't get it compiled into the image. Could you experts take some time to look into it? @M0Rf30 @mudler 

You can't compile cause you need gcc 13. That's why is disabled. You will of course have an error on cLog variable
